spec.md

```diff
@@ -570,6 +572,58 @@ It indicates the type of the data being stored inside of the memory.
 
 See [@sec:mem] for more details.
 
+## Subcomponents
+
+Each circuit component factors into a tree of subcomponents.
+The subcomponents of a circuit component depends on its type and kind.
+
+Subcomponents are themselves circuit components.
+Thus, they have both a kind and a type.
+
+TODO: When we refer to the "circuit components" of a module definition, context will make it clear whether we mean the circuit component declarations (the root circuit components), the leaf circuit components, or the total of all circuit components.
+
+Subcomponents may be named by references (see TODO) and may appear as the target of a connect statement.
+
+We define the tree of circuit components by defining a **direct subcomponent** of a circuit component.
+These are the immediate children of the circuit subcomponent tree.
+
+A **root component** is a circuit component that is not the direct subcomponent of any other component.
+Equivalently, these are the circuit components which are declared in a module.
+
+A **leaf component** is a circuit component has no direct subcomponents.
+Leaf components are useful for checking initialization.
+
+One circuit component is a **subcomponent** of another if there is way to get from the first component to the second through the direct subcomponent relation.
+A circuit component is trivially considered to be a subcomponent of itself.
+If we need to speak of the subcomponents excluding the component itself, we call these the **proper subcomponents**.
+
+When a circuit component has a ground type, an enumeration time, a probe type, or property type (see TODO), has no direct subcomponents.
```

Comment:
```suggestion
When a circuit component has a ground type, an enumeration type, a probe type, or property type (see TODO), has no direct subcomponents.
```

While obvious, does the PR also want to state that there are no proper subcomponents either?

Reply:
A direct subcomponent is always a proper subcomponent. It's so immediate from the definition it shouldn't warrant mention.

However, it seems like I do need to give examples for these terms. They are entirely standard in mathematical discussions, but I should be mindful they take a small bit of working with them to build up an intuition.